control/subscription_group_medium_test.go

```diff
@@ -0,0 +1,671 @@
+			adds, removes := comparePlugins([]core.SubscribedPlugin{plg1, plg2}, []core.SubscribedPlugin{plg1})
+
+			Convey("Plugins to add and plugins to remove have proper elements", func() {
+				So(len(adds), ShouldEqual, 1)
+				So(len(removes), ShouldEqual, 0)
+				So(subscribedPluginsContain(adds, plg2), ShouldBeTrue)
+				So(subscribedPluginsContain(adds, plg1), ShouldBeFalse)
+			})
+		})
+	})
+
+	Convey("Given new collector:plg:1 and old collector:plg:2 and collector:plg:1", t, func() {
+		plg1 := mockSubscribedPlugin{
+			typeName: core.CollectorPluginType,
+			name:     "plg",
+			version:  2,
+			config:   cdata.NewNode(),
+		}
+
+		plg2 := mockSubscribedPlugin{
+			typeName: core.CollectorPluginType,
+			name:     "plg",
+			version:  1,
+			config:   cdata.NewNode(),
+		}
+		Convey("When comparing new and old plugins", func() {
+			adds, removes := comparePlugins([]core.SubscribedPlugin{plg1}, []core.SubscribedPlugin{plg1, plg2})
+
+			Convey("Plugins to add and plugins to remove have proper elements", func() {
+				So(len(adds), ShouldEqual, 0)
+				So(len(removes), ShouldEqual, 1)
+				So(subscribedPluginsContain(removes, plg2), ShouldBeTrue)
+				So(subscribedPluginsContain(removes, plg1), ShouldBeFalse)
+			})
+		})
+	})
+}
+
+func TestSubscriptionGroups_ProcessStaticNegative(t *testing.T) {
+	log.SetLevel(log.DebugLevel)
+	c := New(getTestSGConfig())
+
+	lpe := newLstnToPluginEvents()
+	c.eventManager.RegisterHandler("TestSubscriptionGroups_Process", lpe)
+	c.Start()
+
+	Convey("Loading a mock collector plugn", t, func() {
+		_, err := loadPlg(c, path.Join(os.Getenv("SNAP_PATH"), "plugin", "snap-plugin-collector-mock1"))
+		<-lpe.load
+		So(err, ShouldBeNil)
+
+		Convey("Subscription group created for requested metric with no wildcards", func() {
+			requested := mockRequestedMetric{namespace: core.NewNamespace("intel", "mock", "foo")}
+			subsPlugin := mockSubscribedPlugin{
+				typeName: core.CollectorPluginType,
+				name:     "mock",
+				version:  1,
+				config:   cdata.NewNode(),
+			}
+
+			sg := newSubscriptionGroups(c)
+			So(sg, ShouldNotBeNil)
+			sg.Add("task-id", []core.RequestedMetric{requested}, cdata.NewTree(), []core.SubscribedPlugin{subsPlugin})
+			<-lpe.sub
+			So(len(sg.subscriptionMap), ShouldEqual, 1)
+			group, ok := sg.subscriptionMap["task-id"]
+			So(ok, ShouldBeTrue)
+			So(group, ShouldNotBeNil)
+			So(len(group.plugins), ShouldEqual, 1)
+			So(len(group.metrics), ShouldEqual, 1)
+			So(len(group.requestedMetrics), ShouldEqual, 1)
+			So(subscribedPluginsContain(group.plugins, subsPlugin), ShouldBeTrue)
+
+			Convey("loading another mock", func() {
+				anotherMock := mockSubscribedPlugin{
+					typeName: core.CollectorPluginType,
+					name:     "anothermock",
+					version:  1,
+					config:   cdata.NewNode(),
+				}
+
+				_, err := loadPlg(c, path.Join(os.Getenv("SNAP_PATH"), "plugin", "snap-plugin-collector-anothermock1"))
+				<-lpe.load
+				So(err, ShouldBeNil)
```

Comment:
Should we Check the err here before waiting on the event? If it errors we won't ever receive the event (I think).